src/hotspot/share/gc/g1/g1CollectedHeap.inline.hpp

```diff
@@ -233,7 +233,6 @@ inline bool G1CollectedHeap::is_obj_dead_full(const oop obj) const {
 }
 
 inline bool G1CollectedHeap::is_humongous_reclaim_candidate(uint region) {
-  assert(_hrm.at(region)->is_starts_humongous(), "Must start a humongous object");
```

Comment:
I think I like the semantic requirement that this only be applied to starts-humongous regions.  Unless there is some future change that needs this removal (and not just because of the change to G1FreeHumongousRegionClosure::do_heap_region that I don't like), then I'd prefer it stay.

Reply:
There are no other uses in the near future. However, this method doesn't really require that as a pre-condition, even before this PR.

Reply:
That it doesn't (currently) require that is arguably an implementation detail.  I'm speaking of the intended semantics of the function.  I think it's a query about the region that corresponds to a humongous object.  It's placed near other predicates on objects.  That its argument is a region rather than an oop is an inconsistency that arises from how it's being used.  Maybe it's misplaced here?